src/test/java/MorseCodePresenterTest.java

```diff
@@ -0,0 +1,45 @@
+import org.junit.jupiter.api.Test;
+import static org.mockito.Mockito.*;
+
+class MorseCodePresenterTest
+{
+    private final MorseCodeFrame frame = mock(MorseCodeFrame.class);
+    private final MorseCodeConverter converter = mock(MorseCodeConverter.class);
+    private final MorseCodePresenter presenter = new MorseCodePresenter(frame, converter);
+
+    @Test
+    public void translateToEnglish()
+    {
+        //given
+        String morseExpression = ".... . .-.. .-.. --- /.-- --- .-. .-.. -.. /";
+        frame.setMorseExpressionText(morseExpression);
+        frame.setEnglishExpressionText("");
```

Comment:
Frame is a mock. This has no effect.

Reply:
So then do I need to use the frame at all? Or was it better when I had the getters and the doReturn?

Reply:
You need to verify that the method in the frame is called. But there is no reason to set anything in the frame. Take the method calls out, you will see that your test still passes.